packages/neon_framework/packages/neon_http_client/lib/src/neon_http_client.dart

```diff
@@ -64,9 +65,13 @@ final class NeonHttpClient with http.BaseClient {
         );
       }
 
-      builder.add(
-        CSRFInterceptor(client: baseClient, baseURL: baseURL),
-      );
+      builder
+        ..add(
+          CSRFInterceptor(client: baseClient, baseURL: baseURL),
+        )
+        ..add(
+          AuthenticationInterceptor(),
+        );
```

Comment:
The nextcloud package should not ever use neon_http_client.
This is an HTTP client specifically built for the framework.
I get your point for testing, but everything should be tested as close to reality as possible, so including this interceptor at runtime.

Reply:
Then how do you want to handle the persistence of cookies that is needed for the nextcloud tests?

Reply:
As we tell everyone who wants to use the nextcloud package with cookies: Provide your own custom HTTP client.
Since we have cookie_store this is easy to implement and we only need it for testing.
Relying on stuff from the framework is not ok for this purpose.